### DISCUSSION TITLE
How should we determine the prioritization of work?

### DISCUSSION THREAD

**Discussion by darrelmiller** [OAI/OpenAPI-Specification#2693] [General]
## Problem statement

The community don't have a clear understanding of how to communicate what is important to them in a way that we can easily aggregate that information.
The TSC has no defined way to determine what to work on next.

## Questions
Can we use reactions counts to determine priority?
How do we guide people to put reactions in the right place?

Options for identifying work items that have been prioritized by the working group:
- Milestones
- Labels

Can we provide a way for the community to add labels?

Is the community desire for features aligned with tooling desire for features?

## 
TSC need to determine a process and present something that is clear to the community.  If you have suggestions, feel free to comment on this thread.

**Comment by earth2marsh** [comment]
Are Discussions better suited for discourse, and how would we then use Issues? I can imagine that the work-to-be-done, such as what would eventually be satisfied by a PR, would be an issue that might have started as a discussion (e.g., whether it made sense to do the work and what the shape of a solution might take). A bug or documentation fix would be another obvious use of an issue.

Discussions appear to have a built-in "upvote" mechanism in addition to reactions, though no downvote. ~~Because that happens at the thread level, it seems a potentially better sorting mechanism than just reactions (which may count as aggregate of all comments on an issue?)~~ (edit: seems any post has that up-arrow, I need to read the docs, prob). Discussions don't have assignees, only participants. Discussions also have categories, which might help authors sort topics into high level buckets (admins can be edit to re-categorize if necessary). I'm guessing you can refer to issues/PRs by number, such as today's agenda #2687, in a discussion? 

If we did use Discussions, we should change the categories, such as "Enhancements", "Questions", "Tooling", "Best practices", "Events".

Ah, one more interesting aspect is that the "code of conduct" is ever-present in the Discussions area. That is a different posture than with issues.

**Reply by earth2marsh** [reply]
I adjusted some of the categories to show how we might organize discussions. I created an Announcements category that's restricted to admins and added the SIGs as an example, see #2695 . I think there is a lot of promise in the idea of shifting Issues to only being about specific work and using Discussions to align on what the work should be.

**Reply by earth2marsh** [reply]
Ahh, and I hadn't realized that "upvoting" is anonymous, whereas reactions are always attributed.

**Comment by handrews** [comment]
In this week's TDC meeting, we discussed letting patch and minor releases be community-contribution-driven, with the TSC ensuring that there's at least one good "marquee" feature per minor release.  Moonwalk (the next major version) obviously has its own repo and processes now.  This is being tracked in #3528 and  #3529.  Questions about milestones and labels are being tracked in #3509.

All of those have a lot more activity and are collected under the "Contributor Guidance" project (while discussions cannot be added to projects).  I'm going to close this in favor of those issues/projects, but of course feel free to re-open if you prefer using discussions for this.
